Support/Scripts/tidy.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3.7
+
+import argparse
+import json
+import subprocess
+import os
+import sys
+
+# This script assumes that there is a compile_commands.json in the current
+# directory. If there is not, you must pass 
+# `--ccdb path/to/compile_commands.json`.  
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--ccdb",
+        default="./compile_commands.json",
+        help="the compile commands database for ds2",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main() -> int:
+    args = parse_args()
+    with open(args.ccdb) as ccdb:
+        compile_commands = json.load(ccdb)
+        os.chdir(compile_commands[0]["directory"])
```

Comment:
This assumes the working directory is the same for every command, which isn't necessarily true. You should move the `os.chdir()` inside the loop. If performance becomes an issue you can implement a check that verifies the directory neesd to be changed before actually changing it, but I'm pretty sure that won't be required.

Reply:
It is unless we add subprojects. All files in our ccdb are compiled out of the build directory e.g. `ds2/build`. 

Also, the purpose of the cd here is that `clang-tidy` reads out of `.`. So you need to be in a directory with a ccdb, not necessarily in the directory that the build actually happens in since cmake generates the ccdb with absolute paths to source files and dependencies and only relative paths for the output.

Reply:
yeah but we have subprojects in ds2 already.

If you need to `chdir()` in the directory where the ccdb is, why not do `os.chdir(os.path.dirname(args.ccd))`? It seems better to do explicitely "move to the directory where ccdb is" instead of having implicit behavior like this. Or am I missing something?

Reply:
> yeah but we have subprojects in ds2 already.

Oh yea, that is true. So I was wrong when I said

> It is unless we add subprojects. All files in our ccdb are compiled out of the build directory e.g. ds2/build.

It always uses `ds2/build` as value the "directory" key in the ccdb. And this is confirmed by looking at a ccdb from llvm. So using the value for the directory key in any entry in a ccdb is valid.

That being said, an explicit path via `os.chdir(os.path.abspath(os.path.dirname(args.ccdb)))` is cleaner.